### ISSUE TITLE
Support for DDS dynamic types?

### ISSUE THREAD

**Issue by kyle-figure** [jhelovuo/RustDDS#269]
Is there a plan to support dynamic types in the future?

DDS Spec: https://www.omg.org/spec/DDS-XTypes
EProsima Fastdds implementation: https://fast-dds.docs.eprosima.com/en/latest/fastdds/dynamic_types/dynamic_types.html

**Comment by jhelovuo** [comment]
No, there is no such plan at the moment.

What would be your application of that?

**Comment by kyle-figure** [comment]
My team is using dynamic types to log internal variables in our program for debugging. We currently log on the order of tens of thousands of variables. This application wouldn't be feasible if we needed to specify each variable in an IDL file.

I would be happy to help with the implementation of dynamic types support in RustDDS if I am able to.

**Comment by ariaameri** [comment]
@kyle-figure hi!

I was wondering if you were able to do the implementation of the xtypes? I specifically am looking for XCDR2 (like fastdds: https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/xtypesExtensions.html#datarepresentationid) to provide types that can evolve and are backward compatible. 

I would appreciate any help!

**Comment by kyle-figure** [comment]
> @kyle-figure hi!
> 
> I was wondering if you were able to do the implementation of the xtypes? I specifically am looking for XCDR2 (like fastdds: https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/xtypesExtensions.html#datarepresentationid) to provide types that can evolve and are backward compatible.
> 
> I would appreciate any help!

No, sorry. I am not planning on working on this anymore either.
